lib/kitchen/driver/ec2.rb

```diff
@@ -320,68 +458,69 @@ def self.hostname(server, interface_type=nil)
         end
       end
 
-      def submit_spot
-        spot = request_spot
-        info("Spot instance <#{spot.id}> requested.")
-        info("Spot price is <#{spot.price}>.")
-        spot.wait_for { print '.'; spot.state == 'active' }
-        print '(spot active)'
-
-        # tag assignation on the instance.
-        if config[:tags]
-          connection.create_tags(
-            spot.instance_id,
-            spot.tags
-          )
-        end
-        connection.servers.get(spot.instance_id)
-      end
-
-      # A mapping from config key values to what Fog expects
-      CONFIG_TO_AWS = {
-        :ebs_volume_size => 'Ebs.VolumeSize',
-        :ebs_volume_type => 'Ebs.VolumeType',
-        :ebs_delete_on_termination => 'Ebs.DeleteOnTermination',
-        :ebs_snapshot_id => 'Ebs.SnapshotId',
-        :ebs_device_name => 'DeviceName',
-        :ebs_virtual_name => 'VirtualName'
-      }
-
       def block_device_mappings
-        bdms = config[:block_device_mappings]
-
-        # If they don't provide one, lets give them a default one
+        bdms = config[:block_device_mappings] || []
         if bdms.nil? || bdms.empty?
-          bdms = [{
-            :ebs_volume_type => 'standard',
-            :ebs_volume_size => config[:ebs_volume_size],
-            :ebs_delete_on_termination => config[:ebs_delete_on_termination],
-            :ebs_snapshot_id => nil,
-            :ebs_device_name => config[:ebs_device_name],
-            :ebs_virtual_name => nil
-          }]
+          if config[:ebs_volume_size] || config[:ebs_delete_on_termination] ||
+            config[:ebs_device_name] || config[:ebs_volume_type]
+            # If the user didn't supply block_device_mappings but did supply
+            # the old configs, copy them into the block_device_mappings array correctly
+            bdm = {}
+            if bdm[:ebs_volume_size].nil?
+              bdm[:ebs_volume_size] = config[:ebs_volume_size] || 8
+            end
+            if bdm[:ebs_delete_on_termination].nil?
+              bdm[:ebs_delete_on_termination] = config[:ebs_delete_on_termination] || true
```

Comment:
@fnichol Yes, `config[:ebs_delete_on_termination]` could be false.  Thanks for the catch!  I'll update to use fetch.